src/targets/python/requests.js

```diff
@@ -28,28 +29,74 @@ module.exports = function (source, options) {
   // Set URL
   code.push('url = "%s"', source.fullUrl).blank();
 
+  const headers = source.allHeaders;
+
   // Construct payload
+  let payload;
+  let hasFiles = false;
   let hasPayload = false;
   let jsonPayload = false;
-  if (source.postData.mimeType === 'application/json') {
-    if (source.postData.jsonObj) {
-      code.push('payload = %s', helpers.literalRepresentation(source.postData.jsonObj, opts));
-      jsonPayload = true;
-      hasPayload = true;
-    }
-  } else {
-    const payload = JSON.stringify(source.postData.text);
-    if (payload) {
-      code.push('payload = %s', payload);
-      hasPayload = true;
-    }
+  switch (source.postData.mimeType) {
+    case 'application/json':
+      if (source.postData.jsonObj) {
+        code.push('payload = %s', helpers.literalRepresentation(source.postData.jsonObj, opts));
+        jsonPayload = true;
+        hasPayload = true;
+      }
+      break;
+
+    case 'multipart/form-data':
+      if (source.postData.params) {
+        const files = {};
+        payload = {};
+
+        source.postData.params.forEach(p => {
+          if (p.fileName) {
+            files[p.name] = `open('${p.fileName}', 'rb')`;
+            hasFiles = true;
+          } else {
+            payload[p.name] = p.value;
+            hasPayload = true;
+          }
+        });
+
+        if (hasFiles) {
+          code.push('files = %s', helpers.literalRepresentation(files, opts));
+
+          if (hasPayload) {
+            code.push('payload = %s', helpers.literalRepresentation(payload, opts));
+          }
+
+          // The requests library will only automatically add a `multipart/form-data` header if
+          // there are files being sent. If we're **only** sending form data we still need to send
+          // the boundary ourselves.
```

Comment:
This is a weird one and apparently `requests` only supports `multipart/form-data` requests if they include `files` payloads`. See this example:

```py
import requests

url = "https://httpbin.org/anything"

payload = {"foo": "bar"}
headers = {"Content-Type": "multipart/form-data"}

response = requests.request("POST", url, data=payload, headers=headers)

print(response.text)
```

```
> python3 __tests__/__fixtures__/output/python/requests/multipart-form-data.py                   
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip, deflate", 
    "Content-Length": "7", 
    "Content-Type": "multipart/form-data", 
    "Host": "httpbin.org", 
    "User-Agent": "python-requests/2.27.1", 
    "X-Amzn-Trace-Id": "REDACTED"
  }, 
  "json": null, 
  "method": "POST", 
  "origin": "REDACTED", 
  "url": "https://httpbin.org/anything"
}
```

Yeah we could pass `foo` in as a `files` param, but it's not a file so its identity would be misrepresented in the incoming payload. For these file-less cases we still need to pass the full boundary:

```py
import requests

url = "https://httpbin.org/anything"

payload = "-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"foo\"\r\n\r\nbar\r\n-----011000010111000001101001--\r\n"
headers = {"Content-Type": "multipart/form-data; boundary=---011000010111000001101001"}

response = requests.request("POST", url, data=payload, headers=headers)

print(response.text)
```